### PR TITLE
Restructure Helm related docs

### PR DESCRIPTION
**Pull request by dholbach**
As discussed in #1205 (and #1178), the Helm docs are not as clear as they should be. We have

- the flux chart README
- a get started doc
- an integration doc
- an integration requirements doc
- (and an [example repo](https://github.com/weaveworks/flux-helm-test))

Almost all in different places.

This PR is a draft of how I'd like to change things:

 - one doc for the helm operator (Michael seemed to be in favour of this too)
 - one doc which explains the helm integration
 - all preferably in the same space

As I don't know all the details, it'd be good if we would review the content as well. I just juggled things around to be more condensed.

Thoughts: 
 - Maybe we can explain the Helm integration in another prominent space where we detail the inner workings of Flux?
 - Maybe we can just document how to write the CRD in the `flux-helm-test` repo? That would slim down the docs even further.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,11 +78,13 @@ Get started by browsing through the documentation below:
...
   - [How it works](/site/how-it-works.md)
   - [Considerations regarding installing Flux](/site/installing.md)
+  - [Flux <-> Helm integration](/site/helm-integration.md)
 - Get Started with Flux
   - [Standalone Flux](/site/standalone/installing.md)
-  - [Flux using Helm](/site/helm/get-started.md)
+  - [Flux using Helm](/site/helm-get-started.md)
 - Operating Flux
   - [Using Flux](/site/using.md)
+  - [the Helm Operator](/site/helm-operator.md)
```
Comment (squaremo) [inline README.md]: Ought to start with a capital, like the others.

File: chart/flux/README.md [hunk 2]
```diff
@@ -12,12 +12,37 @@ a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) p
...
+
+ - One repo containing both desired release state information and Charts
+   themselves.
+ - Release state information in the form of Custom Resources manifests is
+   located under a particular path ("releaseconfig" by default; can be
+   overriden).
+ - Charts are colocated under another path ("charts" by default; can be
+   overriden). Charts are subdirectories under the charts path.
+ - Custom Resource namespace reflects where the release should be done.
+   Both the Helm application and its corresponding Custom Resource will
```
Comment (stefanprodan) [inline chart/flux/README.md]: I would replace `Helm application` with `Helm release`

File: site/helm-integration.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+## Example of manifest content
+
+```
+---
+  apiVersion: helm.integrations.flux.weave.works/v1alpha2
+  kind: FluxHelmRelease
+  metadata:
+    name: mongodb
+    namespace:  myNamespace
+    labels:
```
Comment (stefanprodan) [inline site/helm-integration.md]: Remove the labels section, they are not needed anymore.

File: site/helm-integration.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    releaseName: mongo-database
+    values:
+      image: bitnami/mongodb:3.7.1-r1
+```
+
+## Required fields
+
+ - name
+ - namespace
+ - label.chart  ... the same as chartgitpath, with slash replaced with  an underscore
```
Comment (stefanprodan) [inline site/helm-integration.md]: Remove the labels section, they are not needed anymore.

File: site/helm-integration.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+ - chartGitPath ... path (from repo root) to a Chart subdirectory
+
+
+## Optional fields
```
Comment (stefanprodan) [inline site/helm-integration.md]: We should add here the automated annotations (glob, semver, regex).

Single image example:

```
apiVersion: helm.integrations.flux.weave.works/v1alpha2
kind: FluxHelmRelease
metadata:
  name: podinfo-dev
  namespace: dev
  annotations:
    flux.weave.works/automated: "true"
    flux.weave.works/tag.chart-image: glob:dev-*
spec:
  chartGitPath: podinfo
  releaseName: podinfo-dev
  values:
    image: stefanprodan/podinfo:dev-kb9lm91e
```

Multiple images example:

```
apiVersion: helm.integrations.flux.weave.works/v1alpha2
kind: FluxHelmRelease
metadata:
  name: podinfo-prod
  namespace: prod
  annotations:
    flux.weave.works/automated: "true"
    flux.weave.works/tag.init: semver:~1.0
    flux.weave.works/tag.app: regex:^1.2.*
spec:
  chartGitPath: podinfo
  releaseName: podinfo-prod
  values:
    init:
      image: quay.io/stefanprodan/podinfo:1.2.0
    app:
      image: quay.io/stefanprodan/podinfo:1.0.0
```

File: chart/flux/README.md [hunk 6]
```diff
@@ -211,191 +236,4 @@ Update Weave Flux version with:
 helm upgrade --reuse-values flux \
 --set image.tag=1.3.2 \
 weaveworks/flux
-```
-
-### Installing Weave Flux helm-operator and Helm with TLS enabled
```
Comment (dholbach) [inline chart/flux/README.md]: @stefanprodan Are you OK with moving this to a separate section, namely `site/helm-operator.md`?

**Comment by dholbach** [comment]
> Yes, this is definitely needed. I have held off doing it because I'm not sure we have the right design yet. I'm still not sure, but enough people ask about it, that's it's worth doing anyway.

@squaremo Do you want me to file an issue for this?

> ideally not flux-helm-test

@squaremo An issue for that too?

I addressed the other remarks so far and wait for Stefan's go-ahead wrt moving some docs out of the chart README.

**Comment by dholbach** [comment]
Thanks for the review.

I'll file separate issues to track the other bits which are still missing.

**Pull request merged**